src/NexusMods.DataModel/Activities/Activity.cs

```diff
@@ -0,0 +1,206 @@
+using System.Numerics;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using NexusMods.Abstractions.Activities;
+using NexusMods.Abstractions.Values;
+
+namespace NexusMods.DataModel.Activities;
+
+/// <summary>
+/// A concrete implementation of <see cref="IActivity"/>.
+/// </summary>
+/// <param name="monitor"></param>
+/// <param name="group"></param>
+/// <param name="payload"></param>
+internal class Activity(ActivityMonitor monitor, ActivityGroup group, object? payload) : IActivitySource, IReadOnlyActivity
+{
+    //
+    private readonly Subject<DateTime> _reports = new();
+
+    /// <summary>
+    /// The start time of the activity.
+    /// </summary>
+    protected readonly DateTime StartTime = DateTime.UtcNow;
+
+    /// <summary>
+    /// The current status (text name) for the activity.
+    /// </summary>
+    protected (string Template, object[] Arguments) Status;
+
+    /// <summary>
+    /// The activity progress, stored as a double so we can handle clamping and overflow for rounding errors
+    /// </summary>
+    protected double Percentage;
+
+    /// <summary>
+    /// The unique identifier of the activity.
+    /// </summary>
+    public ActivityId Id { get; } = ActivityId.From(Guid.NewGuid());
```

Comment:
This can use the new `ActivityId.NewId()` method that gets added by the source generator for `Guid` VOs.